src/main/java/uk/gov/hmcts/ccd/domain/service/common/StateReferenceService.java

```diff
@@ -0,0 +1,50 @@
+package uk.gov.hmcts.ccd.domain.service.common;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import javax.inject.Inject;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.domain.casestate.EnablingConditionParser;
+import uk.gov.hmcts.ccd.domain.model.definition.EventPostStateDefinition;
+
+@Service
+public class StateReferenceService {
+
+    private final EnablingConditionParser enablingConditionParser;
+
+    @Inject
+    public StateReferenceService(EnablingConditionParser enablingConditionParser) {
+        this.enablingConditionParser = enablingConditionParser;
+    }
+
+    public Optional<String> evaluatePostStateCondition(List<EventPostStateDefinition> eventPostStateDefinitions,
```

Comment:
mmm why Optional? There must always be a post state for a case, from a business point of view is not optional. So modelling this in the API as Optional it feels to me a bit confusing. I'd just define this as String, also considering that we already have import validation that mandates specifying a default case right?

Reply:
Yes.. I will change this

Reply:
changed, now returning String